libs/clients/radom/client.go

```diff
@@ -0,0 +1,107 @@
+package radom
+
+import (
+	"crypto/subtle"
+	"errors"
+	"os"
+
+	"github.com/brave-intl/bat-go/libs/clients"
+	"github.com/shopspring/decimal"
+)
+
+// KeyValue is the structure of the key value pairs
+type KeyValue struct {
+	Key   string                 `json:"key"`
+	Value map[string]interface{} `json:"value"`
+}
+
+// Metadata is the structure of metadata
+type Metadata []KeyValue
+
+// Get allows one to get a value based on key from the radom metadata
+func (rm *Metadata) Get(key string) (map[string]interface{}, error) {
+	for _, v := range []KeyValue(*rm) {
+		if subtle.ConstantTimeCompare([]byte(key), []byte(v.Key)) == 1 {
+			return v.Value, nil
+		}
+	}
+	return nil, errors.New("failed to get key from radom metadata")
+}
+
+// WebhookRequest is the request from radom webhooks
+type WebhookRequest struct {
+	EventName            string   `json:"eventName"`
```

Comment:
The webhook payload structure has changed with the managed payments API. There are 2 webhook messages you can listen on - either for a new subscription or a recurring payment. The recurring payment event would happen every time a charge happens on the subscription period basis - and the new subscription event would only trigger once (and a respective subscription cancellation event would trigger in the event it's cancelled).

```json
{
  "eventType": "managedRecurringPayment",
  "eventData": {
    "managedRecurringPayment": {
      "paymentMethod": {
        "network": "SepoliaTestnet",
        "token": "0x5d684d37922daf7aa2013e65a22880a11c475e25"
      },
      "amount": "10.149736300000000",
      "alternativeMethodPayments": null,
      "transactions": null
    }
  },
  "radomData": {
    "checkoutSession": {
      "checkoutSessionId": "17f3a621-ce07-4d38-883b-3d31622a4c9d",
      "metadata": [
        {
          "key": "brave-metadata",
          "value": "some-id"
        }
      ]
    }
  }
}
```

```json
{
  "eventType": "newSubscription",
  "eventData": {
    "newSubscription": {
      "subscriptionId": "945954fa-badc-498b-9978-7f1712c05cc1",
      "subscriptionType": {
        "automatedEVMSubscription": {
          "buyerAddress": "0x850f184effee1f86472c056b777126bd063e4cd0",
          "subscriptionContractAddress": "0x87Fcf739aEe21FAb3b99121fA4f2D7bF243f9F15",
          "recommendedAllowanceDuration": null
        }
      },
      "network": "SepoliaTestnet",
      "token": "0x5D684d37922dAf7Aa2013E65A22880a11C475e25",
      "amount": "1.9900",
      "currency": "USD",
      "period": "annually",
      "periodCustomDuration": null,
      "createdAt": "2023-07-12T21:20:13.458273Z",
      "tags": {
        "productId": "c33369a9-5afc-4818-9ab3-1793297f1e0e"
      }
    }
  },
  "radomData": {
    "checkoutSession": {
      "checkoutSessionId": "17f3a621-ce07-4d38-883b-3d31622a4c9d",
      "metadata": [
        {
          "key": "brave-metadata",
          "value": "some-id"
        }
      ]
    }
  }
}
```

Apologies for the big changes in the structure. Here is a link to our most recent docs in regards to these webhook messages: https://docs.radom.network/guides/webhooks